test/test_nestedtensor.py

```diff
@@ -482,6 +482,76 @@ def test_clone(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, msg):
             nt1.clone(memory_format=torch.channels_last)
 
+    # cannot test torch.float16 because: RuntimeError: "bernoulli_scalar_cpu_" not implemented for 'Half'
+    @dtypes(torch.float, torch.double)
+    def test_dropout(self, device, dtype):
```

Comment:
Testing looks great & extensive! Out of scope for this PR, but eventually we should think about tying nested tensors in with `OpInfo` somehow. There's a couple ways we could do this:
* Expand the `sample_inputs_func` for supported ops to return samples with nested tensors. Existing generated tests will all be run for these samples, which may or may not be expected to work without a lot of skips.
    *  We may need granular skipping to do it right (see #79161).
* Use the `OpInfo` database to generate our own set of tests within `test_nestedtensor.py`.
    * Generated tests here could use the sample inputs already defined to create nested tensor inputs. One idea credit of @YifanShenSZ is to join the samples into a single nested tensor (this may require a workaround if the sample set includes e.g. no-batch-dim samples).
    * We'd likely need to make `OpInfo` aware of nested tensors (e.g. by adding `supports_nested` to it) to indicate per-op support.

Reply:
another thought: organization-wise, consider separate tests for success cases vs. error cases. You could potentially parametrize the working case over different probability values to reduce some duplication:

```python
from torch.testing._internal.common_utils import parametrize
...
@parametrize("p", [0.0, 0.2, 0.5, 1.0])
def test_dropout_success(self, p):
    # run test with p
    ...
```

Not crucial but may help keep tests succinct and readable as we continue adding op support!

Reply:
another thought: organization-wise, consider separate tests for success cases vs. error cases. You could potentially parametrize the working case over different probability values to reduce some duplication:

```python
from torch.testing._internal.common_utils import parametrize
...
@parametrize("p", [0.0, 0.2, 0.5, 1.0])
def test_dropout_success(self, p):
    # run test with p
    ...
```

Not crucial but may help keep tests succinct and readable as we continue adding op support!